client/ayon_core/tools/attribute_defs/widgets.py

```diff
@@ -313,14 +313,29 @@ def _ui_init(self):
         self.main_layout.addWidget(multisel_widget, 0)
 
     def eventFilter(self, obj, event):
-        if (
+        if event.type() == QtCore.QEvent.Type.ContextMenu:
+            QtCore.QTimer.singleShot(0, self._on_menu_timeout)
+            super(NumberAttrWidget, self).contextMenuEvent(event)
+        elif (
             self._multivalue
             and obj is self._input_widget
             and event.type() == QtCore.QEvent.FocusOut
         ):
             self._set_multiselection_visible(True)
         return False
 
+    def _on_menu_timeout(self):
+        menu = self.findChild(QtWidgets.QMenu, "qt_edit_menu")
```

Comment:
I wonder if all DCCs are using this name of menu, considering Houdini, Nuke, Maya and SubstancePainter etc. Could someone validate those DCCs if the right click does work for text fields there?